test/lib/stack.spec.js

```diff
@@ -0,0 +1,107 @@
+import internal from 'lib/stack-internal.js'
+import client from 'lib/stack-client.js'
+import stack from 'lib/stack.js'
+
+
+const cozyURL = "https://test.mycozy.cloud"
+const token = "mytoken"
+const onCreateApp = function() {}
+const onDeleteApp = function() {}
+
+let internalInit
+let clientInit
+let internalGetApp
+let clientGetApp
+
+describe("stack proxy", () => {
+
+  beforeAll(() => {
+    
+    internalInit = internal.init
+    internal.init = jest.fn( () => Promise.resolve() )
+    internalGetApp = internal.get.app
+    internal.get.app = jest.fn( () => Promise.resolve() )
+
+    clientInit = client.init
+    client.init = jest.fn( () => Promise.resolve() )
+    clientGetApp = client.get.app
+    client.get.app = jest.fn( () => Promise.resolve() )
```

Comment:
Instead of saving references to the functions you are replacing with mocks, you can use 

```
jest.spyOn(client, 'init').mockImplementation(jest.fn())
````

which does the saving for you and provides you with an easy way to restore things afterwards : in the `afterAll` you can write `jest.restoreAllMocks()`

Reply:
Oh! 🎉